### PR TITLE
Proto null fix

### PR DESCRIPTION
**Pull request by gortiz**
This PR fixes the way nulls are read in Protobuf and add some test that verify the correct behavior.

See #11712 to see the same tests failing in current master branch.

### CODE REVIEW

**Comment by gortiz** [comment]
cc @swaminathanmanish

**Review by swaminathanmanish — APPROVED** [review]
Thanks ! Your doc that explains the behavior makes sense
https://docs.google.com/document/d/1xfYVL8V600lPV7U6ZO4NpGIV3gOVp8cTlKn2Zj0bCOI/edit

**Review by walterddr — APPROVED** [review]
looks good to me. minor question. maybe we can copy paste the tl;dr from the google doc or link the google doc here

File: pinot-plugins/pinot-input-format/pinot-protobuf/src/main/java/org/apache/pinot/plugin/inputformat/protobuf/ProtoBufRecordExtractor.java [hunk 1]
```diff
@@ -62,7 +62,7 @@ private Object getFieldValue(Descriptors.FieldDescriptor fieldDescriptor, Messag
     // Note w.r.t proto3 - If a field is not declared with optional keyword, there's no way to distinguish
     // if its explicitly set to a proto default or not been set at all i.e hasField() returns false
     // and we would use null.
-    if (fieldDescriptor.isRepeated() || !fieldDescriptor.isOptional() || message.hasField(fieldDescriptor)) {
+    if (fieldDescriptor.isRepeated() || !fieldDescriptor.hasPresence() || message.hasField(fieldDescriptor)) {
```
[the remark this answers is not in the document]
Comment (gortiz) [reply pinot-plugins/pinot-input-format/pinot-protobuf/src/main/java/org/apache/pinot/plugin/inputformat/protobuf/ProtoBufRecordExtractor.java:65]: Yes, you are right. I'm opening a new PR to change that

**Pull request merged**